qiskit/result/result.py

```diff
@@ -181,6 +182,87 @@ def get_counts(self, experiment=None):
         except KeyError:
             raise QiskitError('No counts for experiment "{0}"'.format(experiment))
 
+    def get_assertion(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+
+        Returns:
+            tuple: tuple containing:
+
+                chisq(float): the chi-squared value
+                pval(float): the p-value
+                passed(Boolean): if assertion passed
+        """
+        exp = experiment
+        counts = self.get_counts(exp)
+        assertion = exp.data[-1][0]
+
+        cbits = assertion._cbit
+        cbits = Asserts.clbits2idxs(cbits, exp)
+
+        new_counts = {}
+        for (key, value) in counts.items():
+            new_key = key.replace(' ', '')
+            new_key = ''.join([new_key[-1*(cbit+1)] for cbit in cbits][::-1])
+            new_counts.setdefault(new_key, 0)
+            new_counts[new_key] += value
+        counts = new_counts
+
+        chisq, pval, passed = assertion.stat_test(counts)
+        return chisq, pval, passed if not assertion._negate else not passed
+
+    def get_assertion_pval(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+        Returns:
+            float: p-value of statistical test
+        """
+        _, pval, _ = self.get_assertion(experiment)
+        return pval
+
+    def get_assertion_chisq(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+
+        Returns:
+            float: chi-squared test statistic
+        """
+        chisq, _, _ = self.get_assertion(experiment)
+        return chisq
+
+    def get_assertion_passed(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+
+        Returns:
+            Boolean: if assertion passed
+        """
+        _, _, passed = self.get_assertion(experiment)
+        return passed
+
+    def get_assertion_type(self, experiment=None):
+        """
+        Return the type of the assertion for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+        Returns:
+            string: the type of assertion
+        """
+        assertion = experiment.data[-1][0]
+        return assertion._type
```

Comment:
if assertions of different types are different classes, the `_type` is unnecessary.

Reply:
The type string is for the user to print out what type of assertion they had run, so given an assertion result (passed or failed), they can easily tell whether it was a "Not Product" assertion or a "Product" assertion.